### PR TITLE
v6: showPayButton defaults to true

### PR DESCRIPTION
**Pull request by sponglord**
## Summary
Update `showPayButton` property to default to `true` for both Components and Drop-in.

Previously it defaulted to `true` for Drop-in and `false` for Components (due legacy reasons, when we didn't have a separate `PayButton` component).

## Tested scenarios
Tested across:
- AmazonPay,
- GooglePay,
- Oxxo,
- IssuerLists,
- PayPal,
- MultiBanco,
- BankTransfer,
- InstantPaymentMethods,
- Pix,
- Redirect,
- Twint,
- Vipps

All of which were setting their own default values for `showPayButton`

Also tested in Dropin, Cards & Components generally

### CODE REVIEW

File: packages/lib/src/components/helpers/QRLoaderContainer.tsx [hunk 1]
```diff
@@ -27,14 +28,15 @@ export interface QRLoaderContainerProps extends UIElementProps {
...
+    // Using the generic here allow to fully extend the QRLoaderContainer (including its props)
     protected static defaultProps = {
         qrCodeImage: '',
         amount: null,
         paymentData: null,
         onError: () => {},
         onComplete: () => {},
-        onActionHandled: () => {}
+        onActionHandled: () => {},
+        showPayButton: false
```
Comment (ribeiroguilherme) [inline packages/lib/src/components/helpers/QRLoaderContainer.tsx:39]: Why `showPayButton: false` here?

File: packages/playground/src/pages/Components/Components.js [hunk 2]
```diff
@@ -97,10 +96,13 @@ getPaymentMethods({ amount, shopperLocale }).then(async paymentMethodsResponse =
     window.upi = checkout.create('upi').mount('.upi-field');
 
     // PIX
-    window.pix = checkout.create('pix', { countdownTime: 5 }).mount('.pix-field');
+    window.pix = checkout.create('pix', { countdownTime: 5, showPayButton: true }).mount('.pix-field');
 
     // Oxxo
-    window.oxxo = checkout.create('oxxo').mount('.oxxo-field');
+    window.oxxo = checkout.create('oxxo', { showPayButton: true }).mount('.oxxo-field');
+
```
Comment (ribeiroguilherme) [inline packages/playground/src/pages/Components/Components.js:103]: Do we need to set `showPayButton: true` here?

File: packages/lib/src/components/Dropin/elements/createInstantPaymentElements.ts [hunk 3]
```diff
@@ -9,6 +9,6 @@ import UIElement from '../../UIElement';
  * @param create - Reference to the main instance `Core#create` method
  */
 const createInstantPaymentElements = (paymentMethods: PaymentMethod[] = [], props, create): Promise<UIElement[]> | [] =>
-    paymentMethods.length ? createElements(paymentMethods, { ...props, isInstantPayment: true, showPayButton: true }, create) : [];
```
Comment (ribeiroguilherme) [inline packages/lib/src/components/Dropin/elements/createInstantPaymentElements.ts:12]: We need the `showPayButton: true` here for Instant payment methods - they are not supposed to be used with custom pay button

**Review by ribeiroguilherme — APPROVED** [review]

File: packages/lib/src/components/helpers/QRLoaderContainer.tsx (hunk 1 above)
Comment (sponglord) [reply packages/lib/src/components/helpers/QRLoaderContainer.tsx:39]: QRCode components, outside of the Dropin, don't need to show a pay button

File: packages/playground/src/pages/Components/Components.js (hunk 2 above)
Comment (sponglord) [reply packages/playground/src/pages/Components/Components.js:103]: This is purely so the Pix component in the components page in the playground isn't just an empty html element.
TBH I don't know why components like these are added to the components page

File: packages/lib/src/components/Dropin/elements/createInstantPaymentElements.ts (hunk 3 above)
Comment (sponglord) [reply packages/lib/src/components/Dropin/elements/createInstantPaymentElements.ts:12]: ok

File: packages/lib/src/components/helpers/QRLoaderContainer.tsx (hunk 1 above)
Comment (ribeiroguilherme) [reply packages/lib/src/components/helpers/QRLoaderContainer.tsx:39]: But we want the showPayButton to default to `true`, right? In this case, it is defaulting to false.

I tested on dev playground, and if I create the Swish component ( `checkout.create('swish')...`). , the Pay button isn't displayed with this change. Removing the default prop make the button show up.

Comment (sponglord) [reply packages/lib/src/components/helpers/QRLoaderContainer.tsx:39]: Ok, I'll set it to true!

File: packages/lib/src/components/BcmcMobile/BcmcMobile.test.ts [hunk 4]
```diff
@@ -20,7 +20,7 @@ describe('BcmcMobile', () => {
     describe('render', () => {
         test('does not render anything by default', () => {
```
Comment (ribeiroguilherme) [inline packages/lib/src/components/BcmcMobile/BcmcMobile.test.ts:21]: @sponglord I think the test descriptions are misleading now, as we are rendering the something by default!

Comment (sponglord) [reply packages/lib/src/components/BcmcMobile/BcmcMobile.test.ts:21]: Doh! I'll fix them!

**Pull request merged**